### PR TITLE
bugfix v2

### PR DESCRIPTION
**Pull request by boeschf**

### CODE REVIEW

File: communication/communication.hpp [hunk 1]
```diff
@@ -42,7 +42,9 @@ template <class CommGrid> vec<std::size_t, 2> offset(CommGrid &&grid) {
   return comm_offset(std::forward<CommGrid>(grid));
 }
 
-template <class CommGrid> auto sub_grid(CommGrid &&grid, int id = 0) {
```
Comment (fthaler) [inline communication/communication.hpp]: I would just change the signature here, i.e. use `CommGrid &grid` (and remove the `std::forward` of course). This should fix the problem without any further intervention, shouldn’t it?

Comment (boeschf) [reply communication/communication.hpp]: unfortunately it does not. I can certainly remove the universal reference. but the problem lies in the `auto` return type - ghex backend returns a (r-)value while the other backends return a l-value reference. `auto` does not deduce the right type.

File: communication/communication.hpp [hunk 2]
```diff
@@ -42,7 +42,9 @@ template <class CommGrid> vec<std::size_t, 2> offset(CommGrid &&grid) {
...
+template <class CommGrid>
+decltype(auto) sub_grid(CommGrid &&grid,
```
Comment (boeschf) [inline communication/communication.hpp]: this fixes all the problems, what do you think @fthaler?

File: communication/communication.hpp (hunk 1 above)
Comment (fthaler) [reply communication/communication.hpp]: Ah sorry, didn’t check closely enough then!

File: communication/communication.hpp (hunk 2 above)
Comment (fthaler) [reply communication/communication.hpp]: That’s perfect. GT requires C++ 14 anyway, so `decltype(auto)` is available.

File: benchmark.cpp [hunk 3]
```diff
@@ -53,7 +53,8 @@ int main(int argc, char **argv) {
 
   // get number of openmp threads
   int num_omp_threads = 1;
-#pragma omp parallel master
+#pragma omp parallel
+#pragma omp master
   num_omp_threads = omp_get_num_threads();
```
Comment (fthaler) [inline benchmark.cpp]: I would prefer `num_omp_threads = omp_get_max_threads()`; without any parallel region here. This should be equivalent (as long as we don’t use `omp_set_num_threads()` or a `num_threads` clause, which AFAIK we don’t).

File: communication/simple_mpi.hpp [hunk 4]
```diff
@@ -26,7 +26,6 @@ struct grid {
   grid &operator=(grid const &) = delete;
   grid &operator=(grid &&);
   ~grid();
-
```
Comment (fthaler) [inline communication/simple_mpi.hpp]: Please undo the change here.
